### PR TITLE
Configure coverage reporting

### PR DESCRIPTION
**Pull request by boolean5**
This PR:

- Configures coverage reporting
- Corrects the order of instructions in the README, to prevent the tests from failing because there is no configuration file.

### CODE REVIEW

**Comment by boolean5** [comment]
This is how I've implemented getting a status badge for the `README`:
1. `pip install coverage-badge` and run `coverage-badge -o coverage.svg` on CircleCI. This is a package that reads the total
  coverage percentage from the `.coverage` file that `coverage.py` produces and creates an svg badge.
2. Upload `coverage.svg` to CircleCI as an artifact
3. In `README.md`, display the `coverage.svg` artifact of the latest CircleCI build of our default branch. When clicked, the badge will direct to the latest HTML coverage report on CircleCI.

File: .circleci/config.yml [hunk 1]
```diff
@@ -23,6 +24,11 @@ jobs:
             source ~/venv/bin/activate
             cp sample_shavar_list_creation.ini shavar_list_creation.ini
             mkdir test-results
-            python -m pytest -v --junitxml=test-results/junit.xml
+            python -m pytest -v --junitxml=test-results/junit.xml --cov=. --cov-report=html --cov-branch
+            coverage-badge -o coverage.svg
```
Comment (say-yawn) [inline .circleci/config.yml:28]: Can we show the coverage report on terminal as well? You should be able to add `--cov-report=term` and it should display the result on the terminal.

File: README.md [hunk 2]
```diff
@@ -2,6 +2,7 @@ shavar-list-creation
 ====================
 
 [![Build Status](https://circleci.com/gh/mozilla-services/shavar-list-creation.svg?style=shield)](https://circleci.com/gh/mozilla-services/shavar-list-creation)
```
[the remark this answers is not in the document]
Comment (say-yawn) [reply README.md:4]: I think we may want to do this for `stage` branch as well and after testing on Stage we can create a PR into `main` branch.

[the remark this answers is not in the document]
Comment (boolean5) [reply README.md]: Done. I made the CircleCI status badge point to `main` branch as we discussed.

File: .circleci/config.yml (hunk 1 above)
Comment (boolean5) [reply .circleci/config.yml]: Good idea. I pushed a commit that does this.

**Pull request merged**